src/store/api.ts

```diff
@@ -19,14 +19,31 @@ export const searchStations = (
 
     const addExpression = (column_name: string, operator: string, values: (Dayjs | string | null)[]) => {
         if (column_name === 'date') {
-            if (values[0]) {
+            if (values[0] && values[1]) {
```

Comment:
One thing I noticed is that the response from API is slower when I hit search with date filter. Could you please add a loading icon preventing users from hitting search again and again?